spring-kafka/src/main/java/org/springframework/kafka/listener/AbstractMessageListenerContainer.java

```diff
@@ -197,8 +197,8 @@ public void setupMessageListener(Object messageListener) {
 	public final void start() {
 		synchronized (this.lifecycleMonitor) {
 			Assert.isTrue(
-					this.containerProperties.getMessageListener() instanceof MessageListener
-							|| this.containerProperties.getMessageListener() instanceof AcknowledgingMessageListener,
+					this.containerProperties.getMessageListener() instanceof IMessageListener
+							|| this.containerProperties.getMessageListener() instanceof IAcknowledgingMessageListener,
```

Comment:
Introducing one more level of abstraction for those generic variants, I won't mind if we add one more: `KafkaDataListener` - a marker interface to avoid similar asserts.
WDYT?